### PR TITLE
[MAIN] [SRVCOM-2744] Disable http2 in webhooks

### PR DESCRIPTION
**Pull request by skonto**
Fixes JIRA SRVCOM-2744

## Proposed Changes

- Disables [http2 for webhooks](https://pkg.go.dev/net/http#hdr-HTTP_2) in webhooks by default to address CVE-2023-44487.

Note: 
a) The controller runtime is a bit old btw needs some update if possible.
b) The pkg stuff are updated based on the latest deps

### CODE REVIEW

File: hack/lib/mesh_resources/authorization-policies/helm/eventing-e2e0.yaml [hunk 1]
```diff
@@ -542,3 +542,21 @@ spec:
         - operation:
             ports:
               - "8022"
+---
```
Comment (skonto) [inline hack/lib/mesh_resources/authorization-policies/helm/eventing-e2e0.yaml:545]: I will get rid of these when the other PR is merged.

File: knative-operator/cmd/manager/main.go [hunk 2]
```diff
@@ -111,6 +112,9 @@ func main() {
 	hookServer.KeyName = "apiserver.key"
 	hookServer.CertName = "apiserver.crt"
 
+	disableHTTP2 := func(c *tls.Config) { c.NextProtos = []string{"http/1.1"} }
```
Comment (skonto) [inline knative-operator/cmd/manager/main.go:115]: Probably we could have a flag here to enable/disable on demand like in other operators.

**Comment by ReToCode** [comment]
We probably also want this here:
https://github.com/openshift-knative/serverless-operator/blob/main/knative-operator/cmd/manager/main.go#L139
and here
https://github.com/openshift-knative/serverless-operator/blob/main/knative-operator/cmd/manager/main.go#L108C25-L108C25
?

**Comment by mgencur** [comment]
Just a note: We probably need to backport this to release-1.30.

**Comment by ReToCode** [comment]
https://github.com/openshift-knative/serverless-operator/pull/2345#issuecomment-1774699121 can be ignored, I checked the servers, and they are fine without changes.

**Comment by skonto** [comment]
Run update-deps with --upgrade to get the latest deps.

**Comment by matzew** [comment]
The updates to eventing / rekt-test, are ok and align the matching 1.11 bits.

For the http/2 there was no special fix, we patched (midstream) the vendor dependency of knative/pkg's `webhook.go` file.

@ReToCode

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

/hold for @pierDipi to veto on this

File: hack/lib/mesh_resources/authorization-policies/helm/eventing-e2e0.yaml [hunk 3]
```diff
@@ -542,21 +542,3 @@ spec:
         - operation:
             ports:
               - "8022"
----
```
Comment (ReToCode) [inline hack/lib/mesh_resources/authorization-policies/helm/eventing-e2e0.yaml:545]: hm that should already be in main: https://github.com/openshift-knative/serverless-operator/commit/bc47ce5976015cf8f7f2509a23727afe0c3f4cc8.

**Comment by skonto** [comment]
> {Failed  === RUN   TestKnativeMetrics/verify_Eventing_metrics_work_correctly
    monitoring_test.go:38: Failed to verify that Eventing control plane metrics work correctly failed to access the Prometheus API endpoint for pingsource_event_count and get the metric value expected: timed out waiting for the condition
    --- FAIL: TestKnativeMetrics/verify_Eventing_metrics_work_correctly (1200.65s)
}

I think we can merge but we need to stabilize that. cc @pierDipi

**Comment by ReToCode** [comment]
I think the error in upstream-e2e-mesh is new, the other ones are tracked [here](https://issues.redhat.com/browse/SRVCOM-2766).

/override "ci/prow/4.11-e2e-kitchensink-ocp-411"

/override "ci/prow/4.13-operator-e2e-aws-ocp-413"

/test 4.11-upstream-e2e-mesh-aws-ocp-411

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

/hold for @pierDipi to veto on this

**Review by pierDipi — APPROVED** [review]
Thanks! 

/lgtm
/approve

**Pull request merged**